src/pkg/packager/filters/deploy_test.go

```diff
@@ -0,0 +1,183 @@
+				}
+
+				name.WriteString(fmt.Sprintf("required=%v", requiredValue))
+
+				if groupValue != "" {
+					name.WriteString(fmt.Sprintf(" && group=%s && idx=%d && default=%t", groupValue, idx, defaultValue))
+				} else if defaultValue {
+					name.WriteString(" && default=true")
+				}
+
+				if groupValue != "" {
+					// if there already exists a component in this group that is default, then set the default to false
+					// otherwise the filter will error
+					defaultAlreadyExists := false
+					if defaultValue {
+						for _, c := range components {
+							if c.DeprecatedGroup == groupValue && c.Default {
+								defaultAlreadyExists = true
+								break
+							}
+						}
+					}
+					if defaultAlreadyExists {
+						defaultValue = false
+					}
+				}
+
+				c := types.ZarfComponent{
+					Name:            name.String(),
+					Default:         defaultValue,
+					DeprecatedGroup: groupValue,
+				}
+
+				if requiredValue != nil {
+					c.Required = helpers.BoolPtr(requiredValue.(bool))
+				}
+
+				components = append(components, c)
+			}
+		}
+	}
+
+	return components
+}
+
+func TestDeployFilter_Apply(t *testing.T) {
+
+	possibilities := componentMatrix(t)
+
+	testCases := map[string]struct {
+		pkg                types.ZarfPackage
+		optionalComponents string
+		want               []types.ZarfComponent
+	}{
+		"Test when version is less than v0.33.0 w/ no optional components selected": {
+			pkg: types.ZarfPackage{
+				Build: types.ZarfBuildData{
+					Version: "v0.32.0",
+				},
+				Components: possibilities,
+			},
+			optionalComponents: "",
+			want: []types.ZarfComponent{
+				cfq(t, "required=<nil> && default=true"),
+				cfq(t, "required=true && default=true"),
+				cfq(t, "required=false && default=true"),
+				cfq(t, "required=true"),
+				cfq(t, "required=<nil> && group=foo && idx=1 && default=true"),
+				cfq(t, "required=<nil> && group=bar && idx=4 && default=true"),
+			},
+		},
+		"Test when version is less than v0.33.0 w/ some optional components selected": {
+			pkg: types.ZarfPackage{
+				Build: types.ZarfBuildData{
+					Version: "v0.32.0",
+				},
+				Components: possibilities,
+			},
+			optionalComponents: strings.Join([]string{"required=false", "required=<nil> && group=bar && idx=5 && default=false", "-required=true"}, ","),
+			want: []types.ZarfComponent{
+				cfq(t, "required=<nil> && default=true"),
+				cfq(t, "required=true && default=true"),
+				cfq(t, "required=false && default=true"),
+				cfq(t, "required=true"),
+				cfq(t, "required=false"),
+				cfq(t, "required=<nil> && group=foo && idx=1 && default=true"),
+				cfq(t, "required=<nil> && group=bar && idx=5 && default=false"),
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// we do not currently support interactive mode in unit tests
+			isInteractive := false
+			filter := ForDeploy(tc.optionalComponents, isInteractive)
+
+			result, err := filter.Apply(tc.pkg)
+			require.NoError(t, err)
```

Comment:
It would be good to test some of the error states that come out of this.

Reply:
done